.github/workflows/service_test_etcd.yml

```diff
@@ -0,0 +1,322 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+name: Service Test Etcd
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+    paths:
+      - "core/src/**"
+      - "core/tests/**"
+      - "!core/src/docs/**"
+      - "!core/src/services/**"
+      - "core/src/services/etcd/**"
+      - ".github/workflows/service_test_etcd.yml"
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}-${{ github.event_name }}
+  cancel-in-progress: true
+
+jobs:
+  etcd-cluster:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Configure Etcd Service
+        # ETCD will use ports: 23790,23791,23792,23800,23801,23802
+        run: |
+            docker network create -d bridge etcd-net
+            docker run -d \
+            -p 23790:2379 \
+            -p 23800:2380 \
+            --network etcd-net \
+            --name etcd1 \
+            -e ALLOW_NONE_AUTHENTICATION=yes \
+            -e ETCD_NAME=etcd1 \
+            -e ETCD_INITIAL_ADVERTISE_PEER_URLS=http://etcd1:2380 \
+            -e ETCD_LISTEN_PEER_URLS=http://0.0.0.0:2380 \
+            -e ETCD_LISTEN_CLIENT_URLS=http://0.0.0.0:2379 \
+            -e ETCD_ADVERTISE_CLIENT_URLS=http://etcd1:2379 \
+            -e ETCD_INITIAL_CLUSTER_TOKEN=etcd-cluster \
+            -e ETCD_INITIAL_CLUSTER=etcd1=http://etcd1:2380,etcd2=http://etcd2:2380,etcd3=http://etcd3:2380 \
+            -e ETCD_INITIAL_CLUSTER_STATE=new \
+            -e ETCD_MAX_REQUEST_BYTES=10485760 \
+            bitnami/etcd:latest
```

Comment:
may be we can use Github Action [service](https://docs.github.com/en/actions/using-containerized-services/about-service-containers) to run the etcd instead of `docker run`. It is easier to understand
```
services:
      etcd:
        image: bitnami/etcd
        ports:
        ...
```